patch_art.sh

```diff
@@ -3,34 +3,105 @@
 set -euo pipefail
 
 this_dir="$(cd "$(dirname "$0")" && pwd)"
-widgets_dir="${this_dir}/zmk/app/boards/shields/nice_view/widgets"
-ART_FILE="${widgets_dir}/art.c"
-PERIPHERAL_STATUS_FILE="${widgets_dir}/peripheral_status.c"
+art_dir="${this_dir}/art"
+config_json="${art_dir}/config.json"
 
-IMG_SIZE="68x140"
-
-reset_zmk() {
-    cd "${this_dir}/zmk"
+magick_args() {
+    local name="$1"
+
+    local config file resize color_convert magick_args
+    config="$(jq -c '.images[] | select(.name == "'"${name}"'") | {name,file,resize,color_convert,magick_args}' "${config_json}")"
+    file="$(         jq -r '.file          // false'  <<<"$config")"
+    resize="$(       jq -r '.resize        // false'  <<<"$config")"
+    color_convert="$(jq -r '.color_convert // "none"' <<<"$config")"
+    magick_args="$(  jq -r '.magick_args   // "-"'    <<<"$config")"
+
+    local src_file="${art_dir}/${file}"
+    local -a args=("${src_file}")
+
+    local IMG_SIZE="68x140"
+    case "$resize" in
+        proportional) # Image is expected to be proportional to 68x140
+            local size_modulus
+            size_modulus=$(file "${src_file}" | perl -p -e 's/.*?\s(\d+)\sx\s(\d+).*/($1\*140)%($2*68)/g' | bc)
+
+            if [[ "$size_modulus" != "0" ]]; then
+                local size
+                size=$(file "${src_file}" | perl -p -e 's/.*?\s(\d+\sx\s\d+).*/$1/g')
+                >&2 echo "ERROR: Image size \(${size}\) not proportional to 68x140. Use either \"ignoreAspectRatio\", or \"fill\" resize strategy instead."
```

Comment:
```suggestion
                >&2 cat <<EOF
ERROR: Image size (${size}) not proportional to 68x140.
Use either "ignoreAspectRatio", or "fill" resize strategy instead.
EOF
```

`\`s are not needed when everything is quoted. Current printed message is:

ERROR: Image size \(180 x 330\) not proportional to 68x140. Use either "ignoreAspectRatio", or "fill" resize strategy instead.

Also, why not change this to `cat` to be consistent with the code below?